streamer/src/nonblocking/quic.rs

```diff
@@ -895,13 +895,34 @@ impl ConnectionTable {
             self.total_size += 1;
             Some((last_update, exit))
         } else {
-            if let Some(connection) = connection {
-                connection.close(
-                    CONNECTION_CLOSE_CODE_TOO_MANY.into(),
-                    CONNECTION_CLOSE_REASON_TOO_MANY,
-                );
+            // Check if there are duplicate peer connections that can be removed.
```

Comment:
This will simply close existing connections from the same port and create a new connection. I think this might has adverse effect on the connection -- the client might be using that. And the client has a mechanism to retry and fill the pool this will just close previous connection and cause communication failures. I am concerned with this the cascading effect -- connection is working and then a new connection try disconnect that. A subsequent retry disconnect that. The end result might be none of the connection can be used for good working. I think it is probably simpler to maintain the old behavior -- fail the new connection.